src/components/shared/Dialogs/ComponentDuplicateDialog.tsx

```diff
@@ -67,6 +49,12 @@ const ComponentDuplicateDialog = ({
     [setClose],
   );
 
+  useEffect(() => {
+    if (newComponent && open) {
+      generateNewDigest(newComponent, newName).then(setNewDigest);
+    }
+  }, [newComponent, open, newName]);
```

Comment:
personally, i prefer `async... await` so something like this:


```suggestion
  useEffect(() => {
    async function generateNewDigest(
      newComponent: HydratedComponentReference | null | undefined,
      newName: string,
    ) {
      if (newComponent) {
        const { digest } = await replaceComponentName(newComponent, newName);
        setNewDigest(digest)
      }
    }

    if (open) {
      generateNewDigest(newComponent, newName);
    }
  }, [newComponent, open, newName]);
```

Reply:
In effect I can't use async/await without penalty of scaffolding a function around it. So I see no benefits in syntax sugar in this specific case. Under the hood it is same promises. Using async/await is beneficial where it saves time/lines improves readability.